cmd/mountgen1.go

```diff
@@ -0,0 +1,243 @@
+
+var gen1Cmd = &cobra.Command{
+	Use:        "mountgen1",
+	Short:      "Mounts Azure Storage ADLS Gen 1 account using SPN auth",
+	Long:       "Mounts Azure Storage ADLS Gen 1 account using SPN auth",
+	SuggestFor: []string{"mntgen1", "gen1 mount"},
+	Args:       cobra.ExactArgs(1),
+	Hidden:     true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		options.MountPath = args[0]
+		options.ConfigFile = configFile
+		parseConfig()
+
+		err := config.Unmarshal(&options)
+		if err != nil {
+			fmt.Printf("Init error config unmarshall [%s]", err)
+			return err
+		}
+
+		err = options.validate(false)
+		if err != nil {
+			fmt.Printf("mountgen1: error invalid options [%v]", err)
+			return err
+		}
+
+		err = config.UnmarshalKey("azstorage", &AzStorageOpt)
+		if err != nil {
+			log.Err("mountgen1: AzStorage config error [invalid config attributes]")
+			return err
+		}
+
+		// not checking ClientSecret since adlsgen1fuse will be reading secret from env variable (ADL_CLIENT_SECRET)
+		if AzStorageOpt.AuthMode == "" && AzStorageOpt.ClientID != "" && AzStorageOpt.TenantID != "" {
+			AzStorageOpt.AuthMode = "spn"
+		}
+
+		err = config.UnmarshalKey("libfuse", &LibFuseOpt)
+		if err != nil {
+			log.Err("mountgen1: Libfuse config error [invalid config attributes]")
+			return err
+		}
+
+		err = config.UnmarshalKey("file_cache", &FileCacheOpt)
+		if err != nil {
+			log.Err("mountgen1: FileCache config error [invalid config attributes]")
+			return err
+		}
+
+		var logLevel common.LogLevel
+		err = logLevel.Parse(options.Logging.LogLevel)
+		if err != nil {
+			fmt.Println("error: invalid log level")
+		}
+
+		err = mountRustFuse()
+		if err != nil {
+			fmt.Printf("Unable to mount Gen1: %s\n", err.Error())
+			return fmt.Errorf("unable to mount Gen1: %s", err.Error())
+		}
+
+		if !generateJsonOnly {
+			err = runAdlsBinary()
+			if err != nil {
+				return fmt.Errorf("unable to run adlsgen1fuse binary: %s", err.Error())
+			}
+		}
+
+		return nil
+	},
+}
+
+// code to generate json file for rustfuse
+func mountRustFuse() error {
+	rustFuseMap := make(map[string]interface{})
+	if strings.ToLower(AzStorageOpt.AuthMode) == "spn" {
+		// adlsgen1fuse will be reading secret from env variable (ADL_CLIENT_SECRET) hence no reason to include this.
+		// rustFuseMap["clientsecret"] = AzStorageOpt.ClientSecret
+
+		rustFuseMap["clientid"] = AzStorageOpt.ClientID
+		rustFuseMap["tenantid"] = AzStorageOpt.TenantID
+		if AzStorageOpt.ActiveDirectoryEndpoint != "" {
+			rustFuseMap["authorityurl"] = AzStorageOpt.ActiveDirectoryEndpoint
+		} else {
+			rustFuseMap["authorityurl"] = "https://login.microsoftonline.com"
+		}
+
+		rustFuseMap["credentialtype"] = "servicePrincipal"
+	} else {
+		log.Err("mountgen1::MountRustFuse : For Gen1 account only SPN auth is supported")
+		return fmt.Errorf("for Gen1 account only SPN auth is supported")
+	}
+
+	rustFuseMap["resourceurl"] = "https://datalake.azure.net/"
+
+	if LibFuseOpt.AttributeExpiration != 0 {
```

Comment:
Why do we do this 0 check?

Reply:
This is done so that we don't add a field for attributeExpiration having 0 value, in the adlsgen1fuse.json file. This json file is passed as parameter to the gen1 binary.
The same logic is used in gen1 mount for v1